### PR TITLE
T12021 celery test results

### PR DESCRIPTION
**Pull request by gctucker**
Tidy up some parts of the code dealing with LAVA test results, skip "lava" test suite data except for boot tests and speed up HTML LAVA log generation which was causing a performance bottleneck.

Note: This depends on https://github.com/kernelci/kernelci-core-staging/pull/61 to have valid build_info.json with each rootfs.
See also: https://github.com/kernelci/kernelci-frontend/pull/39 to stop counting test sets.

### CODE REVIEW

**Comment by gctucker** [comment]
Timing how long it takes to import `simple` test plan results on staging, with master branch:
```
Dec 05 12:44:47 ip-172-30-0-172 kernelci-celery[6143]: [    INFO/MainThread] Task lava-test[83bb4fd5-ad5c-495e-96a8-efa7db7e98c9] succeeded in 3.96509581478s: ObjectId('5c07c829cf3a0f1810e533bb')
```
with this PR:
```
Dec 05 12:35:28 ip-172-30-0-172 kernelci-celery[5122]: [    INFO/MainThread] Task lava-test[4aef777b-7680-4aba-a7a4-de28f1df200e] succeeded in 1.15841176105s: [ObjectId('5c079a76cf3a0f07d290c7dd')]
```
so it's at least 3x faster.  However, 1.16s is still too much for saving a handful of test results.

This duration does increase with the size of the collection in the database, which means it's not efficient at searching for existing entries.  On my laptop, the same request takes about 1s with all the data imported from the staging database whereas it takes around 115ms with my much smaller local database.  It probably needs some indexes to be created with the `test_group_id` in the `test_case` collection, looking into this now.  This may also explain the slow `count()` operations generated from the front-end queries.

File: app/utils/kci_test/__init__.py [hunk 1]
```diff
@@ -521,6 +521,9 @@ def import_and_save_test_cases(group_doc_id, group_name, test_cases,
     if not all((group_doc_id, group_name, test_cases)):
         return ret_code
 
+    utils.LOG.info("Importing {} test cases into group {}".format(
```
Comment (ana) [inline app/utils/kci_test/__init__.py]: I would use the debug level instead of info so these lines won't appear  in the log when running in production.

**Comment by gctucker** [comment]
Also, loading the front-end now got a lot faster, refreshing this page:
https://staging.kernelci.org/test/board/qemu/job/gtucker/

here with the switch from master branch to this PR at 13:27:
```
Dec 05 13:25:12 ip-172-30-0-172 kernelci-backend[6109]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 39166.01ms
Dec 05 13:25:13 ip-172-30-0-172 kernelci-backend[6109]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 40059.62ms
Dec 05 13:25:20 ip-172-30-0-172 kernelci-backend[6109]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 47872.30ms
Dec 05 13:25:28 ip-172-30-0-172 kernelci-backend[6109]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 55412.15ms
Dec 05 13:25:29 ip-172-30-0-172 kernelci-backend[6109]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 56561.28ms
Dec 05 13:25:29 ip-172-30-0-172 kernelci-backend[6109]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 56864.59ms
Dec 05 13:25:30 ip-172-30-0-172 kernelci-backend[6109]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 57774.73ms
Dec 05 13:25:40 ip-172-30-0-172 kernelci-backend[6109]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 67207.22ms
Dec 05 13:25:46 ip-172-30-0-172 kernelci-backend[6109]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 73617.08ms
Dec 05 13:27:54 ip-172-30-0-172 kernelci-backend[8118]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 1090.49ms
Dec 05 13:27:55 ip-172-30-0-172 kernelci-backend[8118]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 1481.15ms
Dec 05 13:27:55 ip-172-30-0-172 kernelci-backend[8118]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 1912.14ms
Dec 05 13:27:55 ip-172-30-0-172 kernelci-backend[8118]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 1902.42ms
Dec 05 13:27:56 ip-172-30-0-172 kernelci-backend[8118]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 2576.83ms
Dec 05 13:27:56 ip-172-30-0-172 kernelci-backend[8118]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 2577.33ms
Dec 05 13:27:56 ip-172-30-0-172 kernelci-backend[8118]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 2628.73ms
Dec 05 13:27:57 ip-172-30-0-172 kernelci-backend[8118]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 3195.67ms
Dec 05 13:27:57 ip-172-30-0-172 kernelci-backend[8118]: [    INFO/MainThread] 200 POST /batch (127.0.0.1) 3751.99ms
```
That's an improvement of about 20x.

File: app/utils/kci_test/__init__.py (hunk 1 above)
Comment (gctucker) [reply app/utils/kci_test/__init__.py]: I actually wanted them to appear in the log, to see how much data is being processed by the server.  I'll put a comment with an example from the server log for a long test suite to see what it looks like.

Comment (gctucker) [reply app/utils/kci_test/__init__.py]: Here's an example with the `v4l2` test plan:
```Dec 05 13:55:08 ip-172-30-0-172 kernelci-backend[8118]: [    INFO/MainThread] 202 POST /callback/lava/test?lab_name=lab-collabora-dev&status=2&status_string=complete (46.235.227.241) 32.45ms
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Received task: lava-test[0a3f1026-58c9-42e0-881c-cafd5b194f59]
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Processing LAVA test data: job 1372968 from lab-collabora-dev
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Generating v4l2 log files in /var/www/images/kernel-ci/gtucker/kernelci-bisect/kernelci-bisect-snapshot-018-3-g7f8476
c6c64f/arm64/defconfig/lab-collabora-dev
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Saving LAVA v2 callback file lava-json-rk3399-gru-kevin.json data in /var/www/images/kernel-ci/gtucker/kernelci-bisec
t/kernelci-bisect-snapshot-018-3-g7f8476c6c64f/arm64/defconfig/lab-collabora-dev
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Importing 5 test cases into group Output-ioctls
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Adding sub-group Output-ioctls to v4l2
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Updating test group 'v4l2' (5c07998dcf3a0f050590c805) with sub-group ID 5c07998dcf3a0f050590c809
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Importing 4 test cases into group Input/Output-configuration-ioctls
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Adding sub-group Input/Output-configuration-ioctls to v4l2
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Updating test group 'v4l2' (5c07998dcf3a0f050590c805) with sub-group ID 5c079991cf3a0f050590c821
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Importing 4 test cases into group Streaming-ioctls_Test-input-0
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Adding sub-group Streaming-ioctls_Test-input-0 to v4l2
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Updating test group 'v4l2' (5c07998dcf3a0f050590c805) with sub-group ID 5c079993cf3a0f050590c831
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Importing 4 test cases into group Allow-for-multiple-opens
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Adding sub-group Allow-for-multiple-opens to v4l2
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Updating test group 'v4l2' (5c07998dcf3a0f050590c805) with sub-group ID 5c079995cf3a0f050590c83d
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Importing 2 test cases into group Required-ioctls
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Adding sub-group Required-ioctls to v4l2
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Updating test group 'v4l2' (5c07998dcf3a0f050590c805) with sub-group ID 5c07998dcf3a0f050590c806
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Importing 3 test cases into group Codec-ioctls-Input-0
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Adding sub-group Codec-ioctls-Input-0 to v4l2
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Updating test group 'v4l2' (5c07998dcf3a0f050590c805) with sub-group ID 5c079990cf3a0f050590c81d
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Importing 6 test cases into group Control-ioctls-Input-0
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Adding sub-group Control-ioctls-Input-0 to v4l2
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Updating test group 'v4l2' (5c07998dcf3a0f050590c805) with sub-group ID 5c079991cf3a0f050590c826
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Importing 6 test cases into group Input-ioctls
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Adding sub-group Input-ioctls to v4l2
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Updating test group 'v4l2' (5c07998dcf3a0f050590c805) with sub-group ID 5c079993cf3a0f050590c836
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Importing 2 test cases into group Debug-ioctls
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Adding sub-group Debug-ioctls to v4l2
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Updating test group 'v4l2' (5c07998dcf3a0f050590c805) with sub-group ID 5c07998ecf3a0f050590c80f
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Importing 10 test cases into group Format-ioctls-Input-0
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Adding sub-group Format-ioctls-Input-0 to v4l2
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Updating test group 'v4l2' (5c07998dcf3a0f050590c805) with sub-group ID 5c07998fcf3a0f050590c812
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Importing 3 test cases into group Buffer-ioctls-Input-0
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Adding sub-group Buffer-ioctls-Input-0 to v4l2
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Updating test group 'v4l2' (5c07998dcf3a0f050590c805) with sub-group ID 5c079992cf3a0f050590c82d
Dec 05 13:55:08 ip-172-30-0-172 kernelci-celery[8152]: [    INFO/MainThread] Task lava-test[0a3f1026-58c9-42e0-881c-cafd5b194f59] succeeded in 0.245718852151s: [ObjectId('5c07998dcf3a0f050590c80
5')]
```

Comment (gctucker) [reply app/utils/kci_test/__init__.py]: If this is too long, then maybe having one line with the total number of test cases and sub-groups would be better.

**Comment by gctucker** [comment]
Importing the results for the `simple` test plan takes about 4s with the master branch and 80ms with this PR.  That's an improvement of about 50x and I think it essentially solves the issue we had.

File: app/utils/kci_test/__init__.py (hunk 1 above)
Comment (gctucker) [reply app/utils/kci_test/__init__.py]: @ana and now:
```
Dec 05 14:45:41 ip-172-30-0-172 kernelci-backend[12963]: [    INFO/MainThread] 202 POST /callback/lava/test?lab_name=lab-collabora-dev&status=2&status_string=complete (46.235.227.241) 33.44ms
Dec 05 14:45:41 ip-172-30-0-172 kernelci-celery[12997]: [    INFO/MainThread] Received task: lava-test[bed78b8e-eaa9-435a-a8b9-f9c8a978a823]
Dec 05 14:45:41 ip-172-30-0-172 kernelci-celery[12997]: [    INFO/MainThread] Processing LAVA test data: job 1372982 from lab-collabora-dev
Dec 05 14:45:41 ip-172-30-0-172 kernelci-celery[12997]: [    INFO/MainThread] Generating v4l2 log files in /var/www/images/kernel-ci/gtucker/kernelci-bisect/kernelci-bisect-snapshot-018-3-g7f8476c6c64f/arm64/defconfig/lab-collabora-dev
Dec 05 14:45:41 ip-172-30-0-172 kernelci-celery[12997]: [    INFO/MainThread] Saving LAVA v2 callback file lava-json-rk3399-gru-kevin.json data in /var/www/images/kernel-ci/gtucker/kernelci-bisect/kernelci-bisect-snapshot-018-3-g7f8476c6c64f/arm64/defconfig/lab-collabora-dev
Dec 05 14:45:41 ip-172-30-0-172 kernelci-celery[12997]: [    INFO/MainThread] Imported 49 tests in 12 groups from v4l2
Dec 05 14:45:41 ip-172-30-0-172 kernelci-celery[12997]: [    INFO/MainThread] Task lava-test[bed78b8e-eaa9-435a-a8b9-f9c8a978a823] succeeded in 0.224836024921s: [ObjectId('5c07998dcf3a0f050590c805')]
```

**Comment by gctucker** [comment]
Tested OK on staging, here's some example results on rk3288-veyron-jaq:
https://staging.kernelci.org/test/board/rk3288-veyron-jaq/job/gtucker/kernel/kernelci-bisect-snapshot-020-1-ge2532d0d95ee/

**Pull request merged**